src/app/panel/auth/LogIn.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react';
+import { TextInput, Button } from '@tremor/react';
+import { APIService } from '@/services/api';
+import { useRouter } from 'next/navigation';
+
+const LogIn: React.FC = () => {
+  const [username, setUsername] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const [isFormValid, setIsFormValid] = useState<boolean>(false);
+  const router = useRouter();
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    console.log(`Username: ${username}, Password: ${password}`);
+    /*
+    Send the form to server, if username or password is wrong, show an error message.
+    */
+    const response = await APIService.signIn(username, password);
+    if (!response.ok) {
+      console.log('Error: ', response);
+      return;
+    }
+    else {
+      router.push('/panel');
+    }
+  };
+
+  const validateForm = (): void => {
+    if (username.length < 1) {
+      setIsFormValid(false);
+    }
+    else if (password.length < 1) {
+      setIsFormValid(false);
+    }
+    setIsFormValid(true);
+  }
```

Comment:
I believe, it would make sense if the front-end application also limited the maximum number of characters in usernames and passwords.

Reply:
True :+1: